jira/resilientsession.py

```diff
@@ -1,119 +1,262 @@
-            error = r.headers["x-authentication-denied-reason"]
-        elif r.text:
-            try:
-                response = json.loads(r.text)
-                if "message" in response:
-                    # Jira 5.1 errors
-                    error = response["message"]
-                elif "errorMessages" in response and len(response["errorMessages"]) > 0:
-                    # Jira 5.0.x error messages sometimes come wrapped in this array
-                    # Sometimes this is present but empty
-                    errorMessages = response["errorMessages"]
-                    if isinstance(errorMessages, (list, tuple)):
-                        error = errorMessages[0]
-                    else:
-                        error = errorMessages
-                # Catching only 'errors' that are dict. See https://github.com/pycontribs/jira/issues/350
-                elif (
-                    "errors" in response
-                    and len(response["errors"]) > 0
-                    and isinstance(response["errors"], dict)
-                ):
-                    # Jira 6.x error messages are found in this array.
-                    error_list = response["errors"].values()
-                    error = ", ".join(error_list)
-                else:
-                    error = r.text
-            except ValueError:
-                error = r.text
+    if not resp.ok:
+        error = parse_error_msg(resp=resp)
+
         raise JIRAError(
             error,
-            status_code=r.status_code,
-            url=r.url,
+            status_code=resp.status_code,
+            url=resp.url,
             request=request,
-            response=r,
+            response=resp,
             **kwargs,
         )
-    # for debugging weird errors on CI
-    if r.status_code not in [200, 201, 202, 204]:
-        raise JIRAError(
-            status_code=r.status_code, request=request, response=r, **kwargs
-        )
-    # testing for the bug exposed on
-    # https://answers.atlassian.com/questions/11457054/answers/11975162
-    if (
-        r.status_code == 200
-        and len(r.content) == 0
-        and "X-Seraph-LoginReason" in r.headers
-        and "AUTHENTICATED_FAILED" in r.headers["X-Seraph-LoginReason"]
-    ):
-        pass
+
+    return True  # if no exception was raised, we have a valid Response
+
+
+def parse_error_msg(resp: Response) -> str:
+    """Parse a Jira Error message from the Response.
+
+    https://developer.atlassian.com/cloud/jira/platform/rest/v2/intro/#status-codes
+
+    Args:
+        resp (Response): The Jira API request's response.
+
+    Returns:
+        str: The error message parsed from the Response. An empty string if no error.
+    """
+    resp_data: Dict[str, Any] = {}  # json parsed from the response
+    parsed_error = ""  # error message parsed from the response
+
+    if resp.status_code == 403 and "x-authentication-denied-reason" in resp.headers:
```

Comment:
probably so but validated on both server and cloud?

Reply:
I think either way it is out of scope for this PR. Hasn't been reported amongst the issues so it probably works as intended.
Amazingly, I couldn't find the endpoint in the Cloud docs page...

Reply:
Maybe it was for older jira versions? 
As the jira docs don't report it anywhere
https://developer.atlassian.com/search/?q=x-authentication-denied-reason&p=1


There is reference to:
https://developer.atlassian.com/search/?q=x-seraph-loginreason&p=1